scalameta/parsers/shared/src/main/scala/scala/meta/internal/parsers/ScalametaParser.scala

```diff
@@ -3114,35 +3118,56 @@ class ScalametaParser(input: Input)(implicit dialect: Dialect) {
         name match {
           case q: Quasi if endTparamQuasi => q.become[Type.Param]
           case _ =>
-            val tparams = typeParamClauseOpt(ownerIsType = true, ctxBoundsAllowed = false)
+            val tparams = typeParamClauseOpt(
+              ownerIsType = true,
+              ctxBoundsAllowed = dialect.allowImprovedBoundsAndGivens
+            )
             val tbounds = typeBounds()
             val vbounds = new ListBuffer[Type]
             val cbounds = new ListBuffer[Type]
             if (ctxBoundsAllowed) {
               @inline
-              def getBound(): Type = {
-                next()
-                currToken match {
-                  case t: Ellipsis => ellipsis[Type](t, 1)
-                  case _ => typ()
-                }
+              def getBound(allowBoundsAlias: Boolean = false): Type = currToken match {
```

Comment:
the method is used in two places only, so there's no prevailing default, thus i'd remove the default value and specify `getBound(allowAlias = xxx)` explicitly. Also, I'd rename the parameter to `allowAlias` as `Bound` is redundant in the context of this method.

Reply:
Renamed and removed default